src/uu/cp/src/cp.rs

```diff
@@ -1721,124 +1961,19 @@ fn copy_file(
     #[cfg(not(unix))]
     let source_is_fifo = false;
```

Comment:
`source_is_fifo` and `source_is_symlink` (a few lines above) can be moved to `handle_copy_mode` because they can be derived from `source_metadata`. This allows you to simplify the signature of `handle_copy_mode`.